// @faustjs/next/dist/cjs/export/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaustProvider = exports.is404 = exports.getNextStaticProps = exports.getNextServerSideProps = exports.logQueries = exports.getClient = exports.withFaust = exports.config = void 0;
var config_1 = require("../config");
Object.defineProperty(exports, "config", { enumerable: true, get: function () { return config_1.config; } });
Object.defineProperty(exports, "withFaust", { enumerable: true, get: function () { return config_1.withFaust; } });
var gqty_1 = require("../gqty");
Object.defineProperty(exports, "getClient", { enumerable: true, get: function () { return gqty_1.getClient; } });
var log_1 = require("../log");
Object.defineProperty(exports, "logQueries", { enumerable: true, get: function () { return log_1.logQueries; } });
var server_1 = require("../server");
Object.defineProperty(exports, "getNextServerSideProps", { enumerable: true, get: function () { return server_1.getNextServerSideProps; } });
Object.defineProperty(exports, "getNextStaticProps", { enumerable: true, get: function () { return server_1.getNextStaticProps; } });
Object.defineProperty(exports, "is404", { enumerable: true, get: function () { return server_1.is404; } });
var components_1 = require("../components");
Object.defineProperty(exports, "FaustProvider", { enumerable: true, get: function () { return components_1.FaustProvider; } });
