// @faustjs/next/dist/cjs/gqty/hooks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAuthHooks = exports.createHooks = void 0;
const useAuth_1 = require("./useAuth");
const useLazyQuery_1 = require("./useLazyQuery");
const useMutation_1 = require("./useMutation");
const usePaginatedQuery_1 = require("./usePaginatedQuery");
const useQuery_1 = require("./useQuery");
const useSubscription_1 = require("./useSubscription");
const useTransactionQuery_1 = require("./useTransactionQuery");
const useHydrateCache_1 = require("./useHydrateCache");
const useCategory_1 = require("./useCategory");
const usePosts_1 = require("./usePosts");
const usePost_1 = require("./usePost");
const usePage_1 = require("./usePage");
const usePreview_1 = require("./usePreview");
const useLogin_1 = require("./useLogin");
const useLogout_1 = require("./useLogout");
function createHooks(useClient) {
    const useQuery = (0, useQuery_1.create)(useClient);
    const useMutation = (0, useMutation_1.create)(useClient);
    return {
        useQuery,
        useLazyQuery: (0, useLazyQuery_1.create)(useClient),
        useMutation,
        usePaginatedQuery: (0, usePaginatedQuery_1.create)(useClient),
        useSubscription: (0, useSubscription_1.create)(useClient),
        useTransactionQuery: (0, useTransactionQuery_1.create)(useClient),
        useHydrateCache: (0, useHydrateCache_1.create)(useClient),
        useCategory: (0, useCategory_1.create)(useQuery),
        usePosts: (0, usePosts_1.create)(useQuery),
        usePost: (0, usePost_1.create)(useQuery),
        usePage: (0, usePage_1.create)(useQuery),
    };
}
exports.createHooks = createHooks;
function createAuthHooks(useClient) {
    const useQuery = (0, useQuery_1.create)(useClient);
    const useAuth = (0, useAuth_1.create)();
    const useMutation = (0, useMutation_1.create)(useClient);
    return {
        useQuery,
        useAuth,
        useLazyQuery: (0, useLazyQuery_1.create)(useClient),
        useMutation,
        usePaginatedQuery: (0, usePaginatedQuery_1.create)(useClient),
        useSubscription: (0, useSubscription_1.create)(useClient),
        useTransactionQuery: (0, useTransactionQuery_1.create)(useClient),
        useHydrateCache: (0, useHydrateCache_1.create)(useClient),
        useCategory: (0, useCategory_1.create)(useQuery),
        usePosts: (0, usePosts_1.create)(useQuery),
        usePost: (0, usePost_1.create)(useQuery),
        usePage: (0, usePage_1.create)(useQuery),
        usePreview: (0, usePreview_1.create)(useAuth, useQuery),
        useLogin: (0, useLogin_1.create)(useMutation),
        useLogout: (0, useLogout_1.create)(),
    };
}
exports.createAuthHooks = createAuthHooks;
