// @faustjs/core/dist/cjs/gqty/schema.generated.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryIdType = exports.PageIdType = exports.PostIdType = void 0;
/** The Type of Identifier used to fetch a single resource. Default is ID. */
var PostIdType;
(function (PostIdType) {
    /** Identify a resource by the Database ID. */
    PostIdType["DATABASE_ID"] = "DATABASE_ID";
    /** Identify a resource by the (hashed) Global ID. */
    PostIdType["ID"] = "ID";
    /** Identify a resource by the slug. Available to non-hierarchcial Types where the slug is a unique identifier. */
    PostIdType["SLUG"] = "SLUG";
    /** Identify a resource by the URI. */
    PostIdType["URI"] = "URI";
})(PostIdType = exports.PostIdType || (exports.PostIdType = {}));
/** The Type of Identifier used to fetch a single resource. Default is ID. */
var PageIdType;
(function (PageIdType) {
    /** Identify a resource by the Database ID. */
    PageIdType["DATABASE_ID"] = "DATABASE_ID";
    /** Identify a resource by the (hashed) Global ID. */
    PageIdType["ID"] = "ID";
    /** Identify a resource by the URI. */
    PageIdType["URI"] = "URI";
})(PageIdType = exports.PageIdType || (exports.PageIdType = {}));
/** The Type of Identifier used to fetch a single resource. Default is ID. */
var CategoryIdType;
(function (CategoryIdType) {
    /** The Database ID for the node */
    CategoryIdType["DATABASE_ID"] = "DATABASE_ID";
    /** The hashed Global ID */
    CategoryIdType["ID"] = "ID";
    /** The name of the node */
    CategoryIdType["NAME"] = "NAME";
    /** Url friendly name of the node */
    CategoryIdType["SLUG"] = "SLUG";
    /** The URI for the node */
    CategoryIdType["URI"] = "URI";
})(CategoryIdType = exports.CategoryIdType || (exports.CategoryIdType = {}));
